src/Analyzer.Reports/ReportsHelper.cs

```diff
@@ -0,0 +1,121 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using Microsoft.Azure.Templates.Analyzer.Types;
+using Microsoft.CodeAnalysis.Sarif;
+
+namespace Microsoft.Azure.Templates.Analyzer.Reports
+{
+    /// <summary>
+    /// 
+    /// </summary>
+    public class ReportsHelper
+    {
+        /// <summary>
+        /// Get all distinct results in an evaluation organized by source location file
+        /// </summary>
+        /// <param name="evaluations">Evaluations to get results for</param>
+        /// <param name="filesToSkip">Files to not include results from</param>
+        /// <returns></returns>
+        public static Dictionary<string, List<(IEvaluation, IList<IResult>)>> GetResultsByFile(
+            IEnumerable<Types.IEvaluation> evaluations,
+            IEnumerable<string> filesToSkip)
+        {
+            return GetResultsByFile(evaluations, filesToSkip, out int _);
+        }
+
+        /// <summary>
+        /// Get all distinct results in an evaluation organized by source location file
+        /// </summary>
+        /// <param name="evaluations">Evaluations to get results for</param>
+        /// <param name="filesToSkip">Files to not include results from</param>
+        /// <param name="passedEvaluations">Out parameter that gives number of passed evaluations</param>
+        /// <returns></returns>
+        public static Dictionary<string, List<(IEvaluation, IList<IResult>)>> GetResultsByFile(
+            IEnumerable<Types.IEvaluation> evaluations,
+            IEnumerable<string> filesToSkip,
+            out int passedEvaluations)
+        {
+            var resultsByFile = new Dictionary<string, List<(IEvaluation, IList<IResult>)>>();
+            passedEvaluations = 0;
+
+            foreach (var evaluation in evaluations)
+            {
+                if (!evaluation.Passed)
+                {
+                    (var actualFile, var failedResults) = GetResultsByFileInternal(evaluation);
+
+                    // a file's results may have already been output if analyzing directory
+                    if (filesToSkip.Contains(actualFile))
```

Comment:
No this is actual to prevent output results for the same file twice if it has multiple references. Without this, the test case "RedisCache template results have multiple references, file results should be deduped" will fail, as the module file will have results for itself as well as its parent file. This is unnecessary after #308 is fixed.